src/Microsoft.AspNetCore.Http/Internal/ServerAddressTemplate.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+
+namespace Microsoft.AspNetCore.Http.Internal
```

Comment:
This seems like a generally useful type. Can we move it out of the `Internal` namespace?